code/model/editableformfields/EditableFormField.php

```diff
@@ -474,10 +474,23 @@ public function canUnpublish($member = null) {
 	public function doPublish($fromStage, $toStage, $createNewVersion = false) {
 		$this->publish($fromStage, $toStage, $createNewVersion);
 
+		$seenIDs = array();
+
 		// Don't forget to publish the related custom rules...
 		foreach ($this->DisplayRules() as $rule) {
+		    $seenIDs[] = $rule->ID;
 			$rule->doPublish($fromStage, $toStage, $createNewVersion);
+			$rule->destroy();
 		}
+
+		// remove any orphans from the "fromStage"
+        $rules = Versioned::get_by_stage('EditableCustomRule', $toStage)
+            ->filter('ParentID', $this->ID)
+            ->exclude('ID', $seenIDs);
+
+		foreach ($rules as $rule) {
+		    $rule->deleteFromStage($toStage);
```

Comment:
Put `$_REQUEST['showqueries'] = 1` and `unset($_REQUEST['showqueries'])` around this line to debug the postgres error.